plugins/backup/veeam/src/main/java/org/apache/cloudstack/backup/veeam/VeeamClient.java

```diff
@@ -160,7 +155,25 @@ protected void setVeeamSshCredentials(String hostIp, String username, String pas
         this.veeamServerPassword = password;
     }
 
+    private void authenticate() {
+        // https://helpcenter.veeam.com/docs/backup/rest/http_authentication.html?ver=95u4
+        final HttpPost request = new HttpPost(apiURI.toString() + "/sessionMngr/?v=v1_4");
+        try {
+            final HttpResponse response = httpClient.execute(request, httpContext);
+            veeamSessionId = response.getFirstHeader(SESSION_HEADER).getValue();
+            checkAuthFailure(response);
+            if (response.getStatusLine().getStatusCode() != HttpStatus.SC_CREATED) {
+                throw new CloudRuntimeException("Failed to create and authenticate Veeam API client, please check the settings.");
+            }
```

Comment:
shouldn't this be changed in order? if a veeamSessionId is found the rest doesn't matter does it?

Reply:
I've refactored this to fail fast initially when the client is created (which authenticates on creation)